gamla/functional_generic.py

```diff
@@ -338,5 +340,55 @@ def bifurcate(*funcs):
 
 def value_to_dict(key: Text):
     return compose_left(
-        functional.wrap_tuple, functional.prefix(key), functional.wrap_tuple, dict,
+        functional.wrap_tuple,
+        functional.prefix(key),
+        functional.wrap_tuple,
+        dict,
     )
+
+
+_R = TypeVar("_R")
+_E = TypeVar("_E")
+
+
+def reduce_curried(
```

Comment:
i would call it simply `reduce` bc in gamla everything is curried.. so it's quite obvious that this as well will be a curried function. but uty

Reply:
the problem is we already have `gamla.reduce` and we use it in an uncurried way (3 variables). So I don't want to break any unknown users by changing the signature to be 2 variables.

wdyt?

Reply:
let's make this function curried and add `elements` to the function arguments, then we can remove the original `gamla.reduce` and it will be functional for old function calls. `reduce` and `reduce_curried` are kinda duplicated right?

Reply:
if I do that how will it handle both async and sync reducers?

Reply:
as discussed on the phone let's keep it this way